pkg/services/serviceaccounts/api/api.go

```diff
@@ -40,9 +43,27 @@ func (api *ServiceAccountsAPI) RegisterAPIEndpoints(
 	auth := acmiddleware.Middleware(api.accesscontrol)
 	api.RouterRegister.Group("/api/serviceaccounts", func(serviceAccountsRoute routing.RouteRegister) {
 		serviceAccountsRoute.Delete("/:serviceAccountId", auth(middleware.ReqOrgAdmin, accesscontrol.EvalPermission(serviceaccounts.ActionDelete, serviceaccounts.ScopeID)), routing.Wrap(api.DeleteServiceAccount))
+		serviceAccountsRoute.Post("/", auth(middleware.ReqOrgAdmin, accesscontrol.EvalPermission(serviceaccounts.ActionCreate, serviceaccounts.ScopeID)), routing.Wrap(api.CreateServiceAccount))
 	})
 }
 
+// POST /api/serviceaccounts
+func (api *ServiceAccountsAPI) CreateServiceAccount(c *models.ReqContext) response.Response {
+	cmd := serviceaccounts.CreateServiceaccountForm{}
+	if err := web.Bind(c.Req, &cmd); err != nil {
+		return response.Error(http.StatusBadRequest, "Bad request data", err)
+	}
+	user, err := api.service.CreateServiceAccount(c.Req.Context(), &cmd)
+	switch {
+	case errors.Is(err, serviceaccounts.ErrServiceAccountNotFound):
+		return response.Error(http.StatusBadRequest, "Failed to create role with the provided name", err)
+	case err != nil:
+		return response.Error(http.StatusInternalServerError, "Failed to create service account", err)
+	}
+
+	return response.JSON(http.StatusCreated, user).SetHeader("Location", fmt.Sprintf("/api/serviceaccounts/%d", user.Id))
```

Comment:
```suggestion
	return response.JSON(http.StatusCreated, user)
```

This is a REST endpoint, what is the purpose of setting "Location"?

Reply:
I really have no idea, more than we do it in access control.

Reply:
If we can't clearly state the reason for doing something, we shouldn't do it.